File: python/resistor-color/resistor_color.py
"""
Lookup the values for color bands on a resistor
"""

band_values = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9,
}

def color_code(color):
    """
    Return the value of a color code.
    color: The color to lookup
    returns: Number - value of color band or ValueError if not in lookup dictionary
    """
    adjusted_color = color.lower().strip()
    if adjusted_color not in band_values:
        raise ValueError(f'Invalid band color: {color}')
    return band_values[adjusted_color]


def colors():
    """
    Return a list of supported band colors, order is not guaranteed to be anything specific.
    """
    return list(band_values.keys())

File: php/resistor-color/ResistorColor.php
<?php
// Resistor color exercise.

declare(strict_types=1);

// Globals are all messed up. I think the test harness including 
// this file is doing the damage.
// Instead of declaring a variable for a global set it in the $GLOBALS super-global instead.
// In this case I can change it to a const. Why don't const need a $ prefix? Got the const thing from
// community solutions.

const resistorColors = [
    "black" => 0,
    "brown" => 1,
    "red" => 2,
    "orange" => 3,
    "yellow" => 4,
    "green" => 5,
    "blue" => 6,
    "violet" => 7,
    "grey" => 8,
    "white" => 9,
];

// Return all of the color bands supported.
// @returns: Array with the color band names supported (lower-case).
function getAllColors(): array 
{  
    return array_keys(resistorColors);
}

// Return the value for a given color band.
// @param $color: The color band (lower-case expected)
// @returns: The value for the color band, or -1 if not found.
function colorCode(string $color): int
{
    if (array_key_exists($color, resistorColors)) {
        return resistorColors[$color];
    }
    return -1;
}
